helusers/jwt.py

```diff
@@ -4,3 +4,60 @@
                                            patch_jwt_settings)
 except ImportError:
     pass
+
+from django.utils.functional import cached_property
+from jose import jwt
+
+from .settings import api_token_auth_settings
+
+
+class JWT:
+    def __init__(self, encoded_jwt):
+        """The constructor checks that a JWT can be extracted from the
+        provided input but it doesn't validate it in any way. If the
+        input is invalid, an exception is raised."""
+        self._encoded_jwt = encoded_jwt
+        self._claims = jwt.get_unverified_claims(encoded_jwt)
+
+    def validate(self, keys, audience):
+        """Verifies the JWT's signature using the provided keys,
+        and validates the claims, raising an exception if anything fails."""
+
+        options = {
+            "require_aud": True,
+            "require_exp": True,
+        }
+
+        jwt.decode(
+            self._encoded_jwt, keys, options=options, audience=audience
+        )
+
+    @property
+    def issuer(self):
+        """Returns the "iss" claim value."""
+        return self.claims["iss"]
+
+    @property
+    def claims(self):
```

Comment:
Is this property still necessary now that the attribute is set in the constructor?

Reply:
Probably not. So renaming `self._claims` to `self.claims` in constructor and removing this property?

Reply:
Yeah!

Reply:
Now I found a difference. A property can be documented so that's it's available as doc string. An attribute in constructor can't.

Reply:
Okay, we can leave it as a property.